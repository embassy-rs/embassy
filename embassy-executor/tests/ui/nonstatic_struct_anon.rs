#![cfg_attr(feature = "nightly", feature(impl_trait_in_assoc_type))]

struct Foo<'a>(&'a ());

#[embassy_executor::task]
async fn task(_x: Foo<'_>) {}

fn main() {}
