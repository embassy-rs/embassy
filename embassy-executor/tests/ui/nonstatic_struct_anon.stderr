error: Arguments for tasks must live forever. Try using the `'static` lifetime.
 --> tests/ui/nonstatic_struct_anon.rs:6:23
  |
6 | async fn task(_x: Foo<'_>) {}
  |                       ^^
